src/job/job-status.entity.ts

```diff
@@ -0,0 +1,17 @@
+import { Job } from '@prisma/client';
+
+export enum JobStatus {
+  PENDING = 'PENDING',
+  PROCESSING = 'PROCESSING',
+  STAGED = 'STAGED',
+  SUCCESS = 'SUCCESS',
+  FAILED = 'FAILED',
+}
+
+export const getJobStatus = async (job: Job): Promise<JobStatus> => {
+  if (job.transactionId === null) return JobStatus.PENDING;
+  if (job.processedAt === null) return JobStatus.PROCESSING;
```

Comment:
Well... Does the situation that `transactionId` is not null, mean `STAGED` instead `PROCESSING`? Or the `PROCESSING` may be ambiguous. 🤔

Reply:
I think so. It will be need some flag that the job processing is started. I will consider adding field for it.